File: C/Lista 3/ex008.c
#include <stdio.h>
#include <stdlib.h>

/*
 * Entrar com nome, idade e sexo de 5 pessoas, e imprimir na tela o nome dos homens maiores de 21 anos.
 */

int main(int argc, char const *argv[])
{
    int age[3], x = 0;
    char sexo[3][1];
    char nome[3][21], homens21[3][21];

    for (int i = 0; i < 5; i++)
    {
        printf("\nDigite o nome da %da pessoa: ", i + 1);
        fflush(stdin);
        gets(&nome[i]);
        printf("Digite a idade da %da pessoa: ", i + 1);
        scanf("%d", &age[i]);
        printf("Digite o sexo da %da pessoa: ", i + 1);
        scanf("%c ", &sexo[i]);
        if ((sexo[i] == 'M' || sexo[i] == 'm') && age[i] > 21)
        {
            homens21[x][0] = nome[i];
            x++;
        }
    }

    printf("nomes:");
    for (int i = 0; i <= x; i++)
    {
        printf("%s", homens21[i]);
    }

    return 0;
}

File: C#/ex008.cs
using System;

namespace Homens21
{
    class Program
    {
        static void Main(String[] args)
        {
            string[,] pessoas = new string[5, 3]; // Matriz que recebe todos os dados
            string[] nomes = new string[5]; // Vetor que recebe somente os nomes dos homens maiores de 21 anos
            int j = 0, x = 0; // Variaveis auxiliares
            
            Console.WriteLine("Insira abaixo os dados das pessoas:");
            for (int i = 0; i < 3; i++)
            {
                Console.Write("Insira o nome da " + (i+1) + " pessoa: "); 
                pessoas[0, j] = Console.ReadLine();

                Console.Write("Insira a idade da " + (i+1) + " pessoa: ");
                pessoas[1, j] = Console.ReadLine();
                    
                Console.Write("Insira o sexo da " + (i+1) + " pessoa: ");
                pessoas[2, j] = Console.ReadLine();
                Console.WriteLine();
                if (int.Parse(pessoas[1, j]) > 21 && (char.Parse(pessoas[2, j]) == 'M' || char.Parse(pessoas[2, j]) == 'm'))
                {
                    nomes[x] = pessoas[0, j];
                    x++;
                }
                j++;
            }
            if (x != 0)
            {
                Console.WriteLine("\nHomens maiores de 21 anos:");
                for (int i = 0; i < x; i++)
                {
                    Console.Write("Nome: " + nomes[i]);
                }
            }
            else
            {
                Console.WriteLine("Nenhum dado inserido bate com o requerido...");
            }
            Console.WriteLine();
        }
    }
}
